airbyte-config/config-models/src/main/resources/type_exploration/airbyte.proto

```diff
@@ -0,0 +1,76 @@
+import "google/protobuf/timestamp.proto";
+
+// =============================== DEFINITIONS ================================
+// types supported by airbyte.
+enum FieldType {
+  STRING = 0;
+  INT = 1;
+  LONG = 2;
+  BOOLEAN = 3;
+  OBJECT = 4;
+  LIST = 5;
+  DATE = 6;
+  TIMESTAMP_WITH_TIMEZONE = 7;
+  TIMESTAMP_WITHOUT_TIMEZONE = 8;
+  TIME_WITH_TIMEZONE = 9;
+  TIME_WITHOUT_TIMEZONE = 10;
+}
+
+// custom struct for each type
+message AirbyteStringField {
+  required string value = 1;
+}
+
+message AirbyteIntField {
+  required int32 value = 1;
+}
```

Comment:
One of the current problems with our types is that we can't represent "NaN" and "Infinity" - would these work here?

Reply:
I'd be surprised if int-typed fields need nan/inf support, that's generally in the realm of float/double (which, assuming protobuf uses standard ieee double, we'd get natively)

I am interested in how we'd support larger numbers though, e.g. int64, or DBs with even larger `numeric`/`decimal` types